src/proxy.rs

```diff
@@ -160,8 +164,30 @@ impl ProxyHandlerWebSocket {
 
         // Stream frontend messages to backend.
         let stream_to_backend = async move {
-            while let Some(Ok(msg)) = frontend_stream.next().await {
-                if let Err(err) = backend_sink.send(MsgTng::from(msg.as_bytes())).await {
+            while let Some(Ok(msg_axm)) = frontend_stream.next().await {
+                // Ugh this is a lot of work that could be avoided if
+                // there were some way to get the wrapped message out
+                // directly, but there isn't...
+                let msg_tng = if msg_axm.is_binary() {
+                    MsgTng::Binary(msg_axm.as_bytes().to_owned())
+                } else if let Some(str) = msg_axm.to_str() {
+                    MsgTng::Text(str.to_owned())
+                } else if msg_axm.is_ping() {
+                    MsgTng::Ping(msg_axm.as_bytes().to_owned())
+                } else if msg_axm.is_pong() {
+                    MsgTng::Pong(msg_axm.as_bytes().to_owned())
+                } else if let Some((code, reason)) = msg_axm.close_frame() {
+                    MsgTng::Close(Some(CloseFrame {
+                        code: code.into(),
+                        reason: Cow::Owned(reason.to_owned()),
+                    }))
+                } else if msg_axm.is_close() {
+                    MsgTng::Close(None)
+                } else {
+                    tracing::error!(msg = ?msg_axm, "Received message is not binary, text, ping, pong, or close");
+                    return;
```

Comment:
Should this `return` or just `continue`?

Reply:
My rationale behind choosing `return` is that dropping or converting an unknown message-type is likely worse than breaking the connection, when code that uses websockets has to be prepared for the connection getting broken at any time anyway.

Reply:
This seems fine, given that normally this branch should be unreachable.